scripts/run.py

```diff
@@ -0,0 +1,487 @@
+            systems.append("maven")
+        if self.has_file_with_ending(project_path, ".sbt"):
+            systems.append("sbt")
+        if self.has_file_with_ending(project_path, "gradle"):
+            systems.append("gradle")
+        if self.has_file_with_ending(project_path, "Vagrantfile"):
+            systems.append("vagrant")
+        return systems
+
+    def is_build_system_supported(self, src_path):
+        project_build_systems = self.get_build_systems(src_path)
+        return any(system in self.config.get("supported_build_systems") for system in project_build_systems)
+
+class Pipeline:
+    __instance = None
+    def get(self, config_file):
+        if Pipeline.__instance is None:
+            Pipeline.__instance = PipelineImpl(config_file)
+        return Pipeline.__instance
+
+    def __call__(self):
+        return self
+
+@task
+def create_project_info(project_path, config_file=None):
+    P = Pipeline().get(config_file)
+    def count_locs(project_path):
+        if not os.path.exists(os.path.join(project_path, "cloc_report.csv")):
+            P.local("cloc --out=cloc_report.csv --csv .", project_path)
+
+        with open(os.path.join(project_path, "cloc_report.csv")) as csv_file:
+            scala_lines = 0
+            total_lines = 0
+            reader = csv.DictReader(csv_file)
+            for line in reader:
+                lines = int(line["code"])
+                total_lines += lines
+                if line["language"] == "Scala":
+                    scala_lines += lines
+        return {"total": total_lines, "scala": scala_lines}
+
+    project_name = os.path.split(project_path)[1]
+    P.info("[Import][%s] Creating project info..." % project_name)
+
+    version = P.local("git describe --always", project_path, verbose=True)
+    commit = P.local("git rev-parse HEAD", project_path)
+    url = P.local("git config --get remote.origin.url", project_path)
+    reponame = url.split('.com/')[1]
+    sloc = count_locs(project_path)
+    repo_info = json.loads(P.local("curl 'https://api.github.com/repos/%s'" % reponame, project_path))
+    gh_stars = repo_info["stargazers_count"] if "stargazers_count" in repo_info else -1
+    build_system = P.get_build_systems(project_path)
+    project_info = {
+       "name": str(project_name),
+       "version": version,
+       "last_commit": commit,
+       "url": url,
+       "total_loc": sloc["total"],
+       "scala_loc": sloc["scala"],
+       "reponame": reponame,
+       "gh_stars": gh_stars,
+       "build_system": '|'.join(build_system)
+    }
+    P.info("[Import][%s] Capture project info: %s" % (project_name, project_info))
+    with open(os.path.join(project_path, "project.csv"), "w") as csv_file:
+        writer = csv.writer(csv_file, delimiter=',')
+        writer.writerow(project_info.keys())
+        writer.writerow(project_info.values())
+    sys.exit(0)
+
+@task
+def import_single(src_path, config_file=None):
```

Comment:
I still have a bit of a problem with this task. I don't think it should be the pipeline that worries about importing the project. I think the target user should be responsible to get the project source - from git or via any other means. The tool starts when data are ready.

It does create a bunch of files, but they can all be placed inside one folder and you can define a clean task that will simply remove that folder.